File: frontend/projects/api/src/lib/models/connection-nudge-dto.ts
/**
 * An invitation to connect — never an obligation. The prompt is rendered
 * server-side so every surface shows identical copy; `phone` powers the
 * sms:/tel: quick actions when present.
 */
export interface ConnectionNudgeDto {
  readonly personId: string;
  readonly name: string;
  readonly relationship: string | null;
  readonly prompt: string;
  readonly phone: string | null;
}

File: backend/src/Shalom.Application/Contracts/ConnectionNudgeDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// An invitation to connect — never an obligation. Carries the rendered
/// prompt so every surface shows identical copy, and the phone so the
/// client can offer sms:/tel: quick actions.
/// </summary>
public record ConnectionNudgeDto(
    Guid PersonId,
    string Name,
    string? Relationship,
    string Prompt,
    string? Phone);
